lib/elixir/lib/task/supervised.ex

```diff
@@ -96,24 +96,51 @@ defmodule Task.Supervised do
         :erlang.raise(:exit, value, __STACKTRACE__)
 
       kind, value ->
-        log(owner, mfa, {log_value(kind, value), __STACKTRACE__})
+        {fun, args} = get_running(mfa)
+
+        :logger.error(
+          %{
+            label: {Task, :terminating},
+            report: %{
+              name: get_from(owner),
+              starter: self(),
+              function: fun,
+              args: args,
+              reason: {log_value(kind, value), __STACKTRACE__}
+            }
+          },
+          %{
+            domain: [:otp, :elixir],
+            error_logger: %{tag: :error_msg},
+            report_cb: &__MODULE__.format_report/2
```

Comment:
We should keep this one private too and make the name consistent between files (format_log or format_report).

Reply:
From what I see, almost all code in OTP uses remote calls to the `report_cb`, probably the reasoning is the same as in `telemetry`, as this is clearer for the VM.

Reply:
We need to have a reasoning though as doing it just in case is not good enough. Especially because it exports a public API that we would need to `@doc false`. So I would:

1. Use local functions for now
2. Verify with the OTP team why they have used remote ones

Can you please check those? Thanks.

Reply:
Using an external function means the whole second argument will be a compile-time literal. A local function can't be optimised like that.